File: python/392_Is-Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0:
            return True
        if len(t) == 0 or len(t) < len(s):
            return False

        ptr_s = 0
        ptr_t = 0

        while ptr_t < len(t):
            if s[ptr_s] == t[ptr_t]:
                ptr_s += 1
            if ptr_s == len(s):
                return True
            ptr_t += 1
        return False

if __name__ == "__main__":
    sol = Solution()
    s = "acb"
    t = "ahbgdc"
    print(sol.isSubsequence(s, t))

File: c++/392_Is-Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.length() > t.length()) return false;
        if(s.length() == 0) return true;

        for(int ptr_s = 0, ptr_t = 0; ptr_t < t.length(); ++ptr_t){

            if(s[ptr_s] == t[ptr_t]){
                ++ptr_s;
                if(ptr_s == s.length()) return true;
            }
        }
        return false;
    }
};